internal/topo/transform/template.go

```diff
@@ -84,7 +90,18 @@
 				d = m
 			}
 			b, err := c.Encode(d)
-			return b, transformed, err
+			if err != nil {
+				return nil, false, fmt.Errorf("fail to encode data %v for error %v", d, err)
+			}
+			mm, ok := d.(map[string]interface{})
```

Comment:
why need encode twice? why need out converter?

Reply:
This is because the converter will record cols, and if you do not use out here, you will get an unwanted field after encode.